test/distributed/org/apache/cassandra/distributed/shared/ClusterUtils.java

```diff
@@ -1352,9 +1384,56 @@ public static List<File> getDirectories(IInstance instance)
         out.add(getCommitLogDirectory(instance));
         out.add(getHintsDirectory(instance));
         out.add(getSavedCachesDirectory(instance));
+        out.add(getJournalDirectory(instance));
+        out.add(getCdcRawDirectory(instance));
         return out;
     }
 
+    /**
+     * Clean up a cluster by stopping all instances, deleting all directories,
+     * and restarting all instances.
+     * <p>
+     * This effectively resets the cluster to a clean state without having to create
+     * a new cluster instance, which is useful for tests that need to start with a 
+     * fresh state between test phases.
+     *
+     * @param cluster the cluster to clean up
+     */
+    public static void cleanup(Cluster cluster)
+    {
+        stopAll(cluster);
+        for (var inst : cluster)
+        {
+            for (var f : getDirectories(inst))
+            {
+                if (f.exists())
+                    f.deleteRecursive();
+            }
+        }
+        for (var inst : cluster)
+            inst.startup();
+    }
+
+    /**
+     * Clean up a single instance by stopping it, deleting all its directories,
+     * and restarting it.
+     * <p>
+     * This effectively resets the instance to a clean state without having to create
+     * a new instance, which is useful for tests that need to start with a fresh state.
+     *
+     * @param inst the instance to clean up
+     */
+    public static void cleanup(IInvokableInstance inst)
```

Comment:
I would personally shorten both comments. I think most our codebase is dense and hard to navigate, plus we have NL before brackets, which makes it hard to fit things on screen. Having longer comments that that do not add on top of the method content makes it harder to navigate.